app/containers/MessageBox/index.js

```diff
@@ -248,6 +288,12 @@ class MessageBox extends Component {
 		if (!equal(nextState.file, file)) {
```

Comment:
Can you think of a better pattern for this than just appending `if` statements? Is there a slight performance improvement by putting all the deep equals calls (`equal(`)) at the very end?

Reply:
We could identify state and props on arrays and compare them, but that's not related to this PR.

About `equal()`, yes, it makes a deep compare, so it should be at the bottom end.
When `shouldComponentUpdate` is triggered, it executes all `if` statements in order to find a `return`.
Primitive values cost a lot less than objects.
It makes no sense to have expensive comparisons at the top if a primitive was changed.

Reply:
To me converting the `if` to concatenated `||` is more clear in terms or intentions and readability. But I guess you are right that it might be out of scope of this PR. Moving the `equals` to the bottom sounds good though.